streamer/src/recvmmsg.rs

```diff
@@ -96,20 +99,14 @@ pub fn recv_mmsg(sock: &UdpSocket, packets: &mut [Packet]) -> io::Result</*num p
     if nrecv < 0 {
         return Err(io::Error::last_os_error());
     }
-    let mut npkts = 0;
-
-    izip!(addrs, hdrs, packets.iter_mut())
-        .take(nrecv as usize)
-        .filter_map(|(addr, hdr, pkt)| {
-            let addr = cast_socket_addr(&addr, &hdr)?.to_std();
-            Some((addr, hdr, pkt))
-        })
-        .for_each(|(addr, hdr, pkt)| {
-            pkt.meta.size = hdr.msg_len as usize;
-            pkt.meta.set_addr(&addr);
-            npkts += 1;
-        });
-    Ok(npkts)
+    let nrecv = usize::try_from(nrecv).unwrap();
```

Comment:
I think this would be a safer against future mis-modification as
```rust
let nrecv = if nrecv < 0 {
    return Err(io::Error::last_os_error());
} else {
   usize::try_from(nrecv).unwrap()
};
```

Reply:
good point. done